src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -203,6 +205,47 @@ def _get_displayable_name(graph_object):
     return graph_object.display_name or ''
 
 
+def _backfill_assignments_for_co_admins(authz_client, assignee=None):
+    co_admins = authz_client.classic_administrators.list('2015-06-01')  # known swagger bug on api-version handling
+    co_admins = [x for x in co_admins if x.properties.email_address]
+    graph_client = _graph_client_factory()
+    if assignee:  # apply assignee filter if applicable
+        try:
+            uuid.UUID(assignee)
+            result = _get_object_stubs(graph_client, [assignee])
+            if not result:
+                return []
+            assignee = _get_displayable_name(result[0]).lower()
+        except ValueError:
+            pass
+        co_admins = [x for x in co_admins if assignee == x.properties.email_address.lower()]
+
+    if not co_admins:
+        return []
+
+    result, users = [], []
+    for i in range(0, len(co_admins), 10):
+        upn_queries = ["userPrincipalName eq '{}'".format(x.properties.email_address) for x in co_admins[i:i + 10]]
+        temp = list(list_users(graph_client.users, query_filter=' or '.join(upn_queries)))
+        users += temp
```

Comment:
Is there a benefit to doing this in groups of 10? It looks to me like there would be better performance if we collected the users all at once.

Reply:
it is a service side limitation for accepting up to 10 query conditions. Sorry for missing out the code comment with the explanation, i am adding it.